packages/web-app-files/src/components/SideBar/Shares/Collaborators/CollaboratorsEditOptions.vue

```diff
@@ -285,7 +321,113 @@ export default {
+        return
+      }
+
+      const { keyCode } = event
+      const isKeyUpEvent = keyCode === 38
+      const isKeyDownEvent = keyCode === 40
+
+      // to cycle through the list of roles only up and down keyboard events are allowed
+      // if this is not the case we can return early and stop the script execution from here
+      if (!isKeyUpEvent && !isKeyDownEvent) {
+        return
+      }
+
+      // if there is only 1 or no roleSelect we can early return
+      // it does not make sense to cycle through it if value is less than 1
+      const roleSelect = this.$refs.roleSelect || []
+
+      if (roleSelect.length <= 1) {
+        return
+      }
+
+      // obtain active role select in following priority chain:
+      // first try to get the focused select
+      // then try to get the selected select
+      // and if none of those applies we fall back to the first role select
+      const activeRoleSelect =
+        roleSelect.find(rs => rs.$el === document.activeElement) ||
+        roleSelect.find(rs => rs.$el.classList.contains('selected')) ||
+        roleSelect[0]
+      const activeRoleSelectIndex = roleSelect.indexOf(activeRoleSelect)
+      const activateRoleSelect = idx => roleSelect[idx].$el.focus()
+
+      // if the event key is arrow up
+      // and the next active role select index would be less than 0
+      // then activate the last available role select
+      if (isKeyUpEvent && activeRoleSelectIndex - 1 < 0) {
+        activateRoleSelect(roleSelect.length - 1)
+
+        return
+      }
+
+      // if the event key is arrow down
+      // and the next active role select index would be greater or even to the available amount of role selects
+      // then activate the first available role select
+      if (isKeyDownEvent && activeRoleSelectIndex + 1 >= roleSelect.length) {
+        activateRoleSelect(0)
+
+        return
+      }
+
+      // the only missing part is to navigate up or down, this only happens if:
+      // the next active role index is greater than 0
+      // the next active role index is less than the amount of all available role selects
+      activateRoleSelect(activeRoleSelectIndex + (isKeyUpEvent ? -1 : 1))
     }
   }
 }
 </script>
+
+<style lang="scss" scoped>
+.files-recipient-role-drop {
+  &-list {
+    background-color: var(--oc-color-swatch-inverse-default);
+    box-shadow: 0 5px 15px rgba(0, 0, 0, 0.08);
+
+    &:hover .files-recipient-role-drop-btn.selected:not(:hover),
+    &:focus .files-recipient-role-drop-btn.selected:not(:focus) {
+      background-color: var(--oc-color-swatch-inverse-default);
+      color: var(--oc-color-swatch-passive-default);
+
+      ::v-deep .oc-icon > svg {
+        fill: var(--oc-color-swatch-passive-default);
+      }
+    }
+  }
+
+  &-btn {
+    border-radius: 0;
+    width: 100%;
+
+    &:hover,
+    &:focus,
+    &.selected {
+      background-color: var(--oc-color-swatch-primary-hover);
```

Comment:
I think it may be nicer to use the same dark background blue as in the OcSelect component, for both UX (recognition) and A11Y (the swatch-primary-{default, hover} have proven themselves problematic here)
@kulmann @tbsbdr opinions?

Reply:
Hmm, I thought the contrast issues have been solved in all cases and haven't even tested it in that aspect... shoot. Well, my thought was that the brand colour shouldn't have anything to do with interactions and an interaction dedicated colour should be used instead.